plugins/Kaleidoscope-MouseKeys/src/kaleidoscope/plugin/MouseKeys.cpp

```diff
@@ -92,31 +93,27 @@ EventHandlerResult MouseKeys::onNameQuery() {
 
 // -----------------------------------------------------------------------------
 EventHandlerResult MouseKeys::onSetup(void) {
-  kaleidoscope::Runtime.hid().mouse().setup();
   kaleidoscope::Runtime.hid().absoluteMouse().setup();
 
   return EventHandlerResult::OK;
 }
 
 // -----------------------------------------------------------------------------
 EventHandlerResult MouseKeys::afterEachCycle() {
-  // Check timeout for accel update interval.
-  if (Runtime.hasTimeExpired(accel_start_time_, accelDelay)) {
-    accel_start_time_ = Runtime.millisAtCycleStart();
-    // `accel_step` determines the movement speed of the mouse pointer, and gets
-    // reset to zero when no mouse movement keys is pressed (see below).
-    if (mousekeys::wrapper.accel_step < 255 - accelSpeed) {
-      mousekeys::wrapper.accel_step += accelSpeed;
-    }
-  }
+  if (directions_ == 0)
+    return EventHandlerResult::OK;
```

Comment:
> what happens when `directions_` is zero, but, say, `wheel_update_interval_` expires? Wouldn't we fail to send a mouse wheel report then?

We _shouldn't_ send a wheel report then, because no wheel keys are pressed if `directions_ == 0`.  That variable gets updated to reflect the set of wheel (and cursor) movement keys in response to any mouse key event.  I consolidated the two sets of directions because we only need eight bits.  If you think it implies too strongly that `directions_` only contains cursor movement info, perhaps we could rename it, or split it apart into two bitfields; the cost isn't huge, but bot RAM and PROGMEM increase in size, if I recall correctly.

Reply:
> If it is handled elsewhere too, why do we handle it in this event handler aswell?

This handler is the one that gets called every cycle, so this is where we need to send the timeout-based reports.  The `KeyEvent` handlers deal with setting the bits in `directions_` (and send reports immediately because we can't wait for the cycle to end in the case of a Macros macro sending multiple events before the end of a cycle).

Reply:
Makes sense, and indeed, `directions_` is properly documented too, I just didn't read it during review.